src/content/en/updates/2018/12/class-fields.md

```diff
@@ -0,0 +1,214 @@
+prevent people from accessing the property directly.
+
+## Private class fields
+
+That’s where private class fields come in. The new private fields syntax is similar to public
+fields, except [you mark the field as being private by using
+`#`](https://github.com/tc39/proposal-class-fields/blob/master/PRIVATE_SYNTAX_FAQ.md). You can think
+of the `#` as being part of the field name:
+
+```js
+class IncreasingCounter {
+  #count = 0;
+  get value() {
+    console.log('Getting the current value!');
+    return this.#count;
+  }
+  increment() {
+    this.#count++;
+  }
+}
+```
+
+Private fields are not accessible outside of the class body:
+
+```js
+const counter = new IncreasingCounter();
+counter.#count;
+// → SyntaxError
+counter.#count = 42;
+// → SyntaxError
+```
+
+## Public and static properties
+
+Class fields syntax can be used to create public and private static properties and methods as well:
+
+```js
+class FakeMath {
+  // `PI` is a static public property.
+  static PI = 22 / 7; // Close enough.
+
+  // `#totallyRandomNumber` is a static private property.
+  static #totallyRandomNumber = 4;
+
+  // `#computeRandomNumber` is a static private method.
+  static #computeRandomNumber() {
+    return FakeMath.#totallyRandomNumber;
+  }
+
+  // `random` is a static public method (ES2015 syntax)
+  // that consumes `#computeRandomNumber`.
+  static random() {
+    console.log('I heard you like random numbers…')
+    return FakeMath.#computeRandomNumber();
+  }
+}
+
+FakeMath.PI;
+// → 3.142857142857143
+FakeMath.random();
+// logs 'I heard you like random numbers…'
+// → 4
+FakeMath.#totallyRandomNumber;
+// → SyntaxError
+FakeMath.#computeRandomNumber();
+// → SyntaxError
+```
+
+## Simpler subclassing
+
+The benefits of the class fields syntax become even clearer when dealing with subclasses that
+introduce additional fields. Imagine the following base class `Animal`:
+
+```js
+class Animal {
+  constructor(name) {
+    this.name = name;
+  }
+}
+```
+
+To create a `Cat` subclass that introduces an additional instance property, you’d previously have to
+call `super()` to run the constructor of the `Animal` base class before creating the property:
+
+```js
+class Cat extends Animal {
+  constructor(name) {
+    super(name);
+    this.likesBaths = false;
+  }
+  meow() {
+    console.log('Meow!');
+  }
+}
+```
+
+That’s a lot of boilerplate just to indicate that cats don’t enjoy taking baths. Luckily, the class
+fields syntax removes the need for the whole constructor, including the awkward `super()` call:
+
+```js
+class Cat extends Animal {
+  likesBaths = false;
+  meow() {
+    console.log('Meow!');
+  }
+}
+```
+
+## Conclusion
+
+Public class fields are shipping in V8 v7.2 and Chrome 72. We plan on shipping private class fields
+soon.
+
+Questions about this new feature? Comments about this article? Feel free to ping me on Twitter via
+[@mathias](https://twitter.com/mathias)!
+
```

Comment:
Please add the was this helpful widget: `{% include "web/_shared/helpful.html" %}`

Reply:
Done!